common/src/main/scala/org/apache/celeborn/common/meta/WorkerPartitionLocationInfo.scala

```diff
@@ -88,12 +88,12 @@ class WorkerPartitionLocationInfo extends Logging {
       locations: util.List[PartitionLocation],
       partitionInfo: PartitionInfo): Unit = {
     if (locations != null && locations.size() > 0) {
-      partitionInfo.putIfAbsent(
+      partitionInfo.computeIfAbsent(
         shuffleKey,
-        JavaUtils.newConcurrentHashMap[String, PartitionLocation]())
+        (k: String) => { return JavaUtils.newConcurrentHashMap[String, PartitionLocation]() })
```

Comment:
Is this allowed by Java syntax?
```suggestion
        (k: String) => JavaUtils.newConcurrentHashMap[String, PartitionLocation]())
```